packages/block-library/src/footnotes/index.php

```diff
@@ -83,136 +83,130 @@ function register_block_core_footnotes() {
 }
 add_action( 'init', 'register_block_core_footnotes' );
 
-add_action(
-	'wp_after_insert_post',
-	/**
-	 * Saves the footnotes meta value to the revision.
-	 *
-	 * @since 6.3.0
-	 *
-	 * @param int $revision_id The revision ID.
-	 */
-	static function( $revision_id ) {
-		$post_id = wp_is_post_revision( $revision_id );
-
-		if ( $post_id ) {
+/**
+ * Saves the footnotes meta value to the revision.
+ *
+ * @since 6.3.0
+ *
+ * @param int $revision_id The revision ID.
+ */
+function wp_save_footnotes_meta( $revision_id ) {
+	$post_id = wp_is_post_revision( $revision_id );
+
+	if ( $post_id ) {
+		$footnotes = get_post_meta( $post_id, 'footnotes', true );
+
+		if ( $footnotes ) {
+			// Can't use update_post_meta() because it doesn't allow revisions.
+			update_metadata( 'post', $revision_id, 'footnotes', $footnotes );
+		}
+	}
+}
+add_action( 'wp_after_insert_post', 'wp_save_footnotes_meta' );
+
+/**
+ * Keeps track of the revision ID for "rest_after_insert_{$post_type}".
+ *
+ * @since 6.3.0
+ *
+ * @param int $revision_id The revision ID.
+ */
+function wp_keep_revision_id( $revision_id ) {
```

Comment:
Same here re: specificity. Maybe `wp_footnotes_cache_latest_revision_id` or something with "footnotes" 😄

Reply:
yeah makes sense to add "footnotes" to all of them